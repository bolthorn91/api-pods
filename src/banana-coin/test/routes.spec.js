/* global expect, describe, it */
const routes = require('../routes')
const stubs = require('./stubs')


jest.mock('../../services/repository', () => () => {
    const { fake } = require('./stubs')
    return {
/*    create: jest.fn(() => fake),
      getAll: jest.fn(() => fake),
      getByID: jest.fn(() => fake),
      purge: jest.fn(() => fake),
      remove: jest.fn(() => fake),
      update: jest.fn(() => fake), */

      post: jest.fn(() => fake),
      read: jest.fn(() => fake),
      getRegistry: jest.fn(() => fake),
      saveRegistry: jest.fn(() => fake)
      
    }
})

describe('operations routes', () => {
  it('should return hello world', async () => {
    const req = stubs.createReq()
    const res = stubs.createRes()

    await routes.hello(req, res)

    expect(res.send).toHaveBeenCalledWith('Hello world!')
  })

    //get all banana coins test
    it('return all banana coins', async () => {
      const req = stubs.createReq()
      const res = stubs.createRes()
      
      await routes.post(req, res)
  
      expect(res.send).toMatchSnapshot(stubs.createJSONAPI())
    })
    


    //create banana test
    it('creates a new banana coin', async () => {
      const req = stubs.createReq()
      const res = stubs.createRes()
      
      await routes.read(req, res)
  
      expect(res.send).toMatchSnapshot(stubs.createJSONAPI())
    })


})


