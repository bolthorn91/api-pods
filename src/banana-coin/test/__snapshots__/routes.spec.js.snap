// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operations routes creates a new banana coin: [object Object] 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      1337,
    ],
  ],
}
`;

exports[`operations routes return all banana coins: [object Object] 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      1337,
    ],
  ],
}
`;
